objectherkenning_openbare_ruimte/data_sampling/decos_sampling.py

```diff
@@ -0,0 +1,318 @@
+import argparse
+import os
+import pathlib
+import shutil
+import sys
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import shapely.geometry as sg
+
+sys.path.append(os.getcwd())
+
+from objectherkenning_openbare_ruimte.data_sampling import decos_helper  # noqa: E402
+from objectherkenning_openbare_ruimte.settings.settings import (  # noqa: E402
+    ObjectherkenningOpenbareRuimteSettings,
+)
+
+config_path = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), "..", "..", "config.yml")
+)
+
+ObjectherkenningOpenbareRuimteSettings.set_from_yaml(config_path)
+settings = ObjectherkenningOpenbareRuimteSettings.get_settings()
+
+RD_CRS = "EPSG:28992"  # CRS code for the Dutch Rijksdriehoek coordinate system
+LAT_LON_CRS = "EPSG:4326"  # CRS code for WGS84 latitude/longitude coordinate system
+
+
+class DecosSampling:
+    """
+    Class used for sampling image frames based on permit data. A sampling weight
+    parameter can be used to determine the amount of non-permit frames allowed.
+    """
+
+    def __init__(
+        self,
+        input_folder: str,
+        metadata_folder: str,
+        decos_folder: str,
+        output_folder: str,
+    ):
+        """
+        Construct a DataSampling object. The construction immediately populates
+        a dataframe with all images in the input_folder, and adds metadata and
+        permit information.
+
+        After construction, call .run_sampling_job() to perform the data sampling.
+
+        Parameters
+        ----------
+        input_folder: str
+            Path to folder with image frames
+        metadata_folder: str
```

Comment:
can we rename "metadata" to "frame_metadata"? 

it's more explicit. we also have detection metadata, decos metadata, etc